cmd/installer/cmd/iso.go

```diff
@@ -29,7 +29,7 @@ terminal_output console
 menuentry "Talos ISO" {
 	set gfxmode=auto
 	set gfxpayload=text
-	linux /boot/vmlinuz page_poison=1 slab_nomerge slub_debug=P pti=on panic=0 consoleblank=0 earlyprintk=ttyS0 console=tty0 console=ttyS0 talos.platform=metal
+	linux /boot/vmlinuz init_on_alloc=1 init_on_free=1 slab_nomerge pti=on panic=0 consoleblank=0 earlyprintk=ttyS0 console=tty0 console=ttyS0 talos.platform=metal
```

Comment:
Not that it should be blocked for it, but do we know what kind of performance hit this double-init causes?

Reply:
Good question. I see https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1862822. Looks related to ZFS.

Reply:
Also https://github.com/openzfs/zfs/issues/9910.

Reply:
https://cateee.net/lkddb/web-lkddb/INIT_ON_FREE_DEFAULT_ON.html warns of performance issues too. I don't know what the answer is if we want to be kspp compliant :thinking: . Do we just leave the old `page_poison` and `slab_*` options?

Reply:
I guess old options are not much better in terms of performance, but that's my guess

Reply:
@andrewrynhard we talked a bit about it, open option is to skip checking for `init_on_alloc` being set to 1, so that users can disable it via command line args if they need to (but default is still on)

Reply:
👍

Reply:
in fact, `init_on_free` has more performance hit, so we could potentially allow disabling it as well for those who care about performance more than security

Reply:
oh, I just realized the way our installer works, it always enforces these options (as it appends them to the command line last), so there's no easy way to disable them except for mounting `/boot` and editing grub config

Reply:
Sounds like we should make this whole optional thing in 0.9

Reply:
one way around is to let people override defaults with extra kernel args